test/python/compiler/test_assembler.py

```diff
@@ -401,12 +401,29 @@ def test_assemble_meas_map(self):
 
     def test_assemble_memory_slots(self):
         """Test assembling a schedule and inferring number of memoryslots."""
+        acquire_channels = [pulse.AcquireChannel(idx) for idx in range(10)]
+        mem_slots = [pulse.MemorySlot(idx) for idx in range(10)]
```

Comment:
I believe the original test for sparse memory_slot creation was fine, it just had an off by one error in the ground truth. Either keep the original updated test or the test you added below. If the test doesn't add anything (test new features) there is no point in adding extra tests. Thank you!

Reply:
because `max_memory_slot` is updated repeatedly, adding auxiliary test for multiple acquisition is safer for future code update. I updated original test, but added one for multiple acquisition as well.